### PR TITLE
PRO-1448 Initial Set up Animation

### PR DESCRIPTION
**Pull request by lalitkumarrock**
https://linear.app/pillarproject/issue/PRO-1448/pillar-wallet-initial-set-up-animation

## Description

- Introduced animation loader with processing status in initial setup.

## How Has This Been Tested?

- Tested on Android Device
- Tested on Simulator 14 pro

## Screenshots (if appropriate):
For new user:
https://github.com/pillarwallet/pillarwallet/assets/82652040/82b059af-09ad-4aaa-82a6-d294a801a811

For existing user:
https://github.com/pillarwallet/pillarwallet/assets/82652040/d0572621-5454-4483-8f4f-8c2324473be8

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Review by IAmKio — CHANGES_REQUESTED** [review]
Just a few small comments but other than that - looking good

File: src/components/SkeletonLoader/OnboardingLoaders.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+      width={width * 1.05}
+      height={height * 1.05}
+    >
+      <Rect
+        x="0"
+        y="0"
+        rx={radius ? JSON.stringify(radius) : '4'}
+        ry={radius ? JSON.stringify(radius) : '4'}
+        width={JSON.stringify(width)}
+        height={JSON.stringify(height)}
```
[the remark this answers is not in the document]
Comment (lalitkumarrock) [reply src/components/SkeletonLoader/OnboardingLoaders.tsx:55]: Yes, radius, width and height is hardcoded. here made just common component.

File: src/actions/onboardingActions.js [hunk 2]
```diff
@@ -399,7 +418,11 @@ export const setupAppServicesAction = (privateKey: ?string) => {
     logBreadcrumb('onboarding', 'onboardingAction.js: checking user is online or not');
 
     // all the calls below require user to be online
-    if (!isOnline) return;
+    if (!isOnline) {
+      dispatch({ type: SET_FETCHING, payload: false });
+      dispatch({ type: SET_LOADING_MESSAGE, payload: '' });
+      return;
+    }
```
[the remark this answers is not in the document]
Comment (lalitkumarrock) [reply src/actions/onboardingActions.js:425]: It's stuck on particular event.

https://github.com/pillarwallet/pillarwallet/assets/82652040/1b01fa5a-c419-462b-85e3-dd46619cb964

File: src/actions/onboardingActions.js [hunk 3]
```diff
@@ -357,12 +368,20 @@ export const walletSetupAction = (enableBiometrics?: boolean) => {
...
     logBreadcrumb('onboarding', 'walletSetupAction: completed, dispatching SET_FINISHING_ONBOARDING');
     isLogV2AppEvents() && dispatch(logEventAction('v2_account_sign_up_completed'));
     dispatch({ type: SET_FINISHING_ONBOARDING, payload: false });
+
+    dispatch({ type: SET_LOADING_MESSAGE, payload: t('onboardingLoaders.ready') });
+
+    setTimeout(() => {
+      dispatch({ type: SET_FETCHING, payload: false });
+      dispatch({ type: SET_LOADING_MESSAGE, payload: '' });
+    }, 0);
```
[the remark this answers is not in the document]
Comment (lalitkumarrock) [reply src/actions/onboardingActions.js:384]: Yes, you are right. But somehow it takes 3 to 4 seconds to dismiss the Ready! text. It takes longer due to many background processes.

[the remark this answers is not in the document]
Comment (IAmKio) [reply src/actions/onboardingActions.js:384]: Ahh okay, understood!

File: src/actions/onboardingActions.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (IAmKio) [reply src/actions/onboardingActions.js:425]: We should probably handle this a bit better... can we retry the process with a backoff?

File: src/components/SkeletonLoader/OnboardingLoaders.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (IAmKio) [reply src/components/SkeletonLoader/OnboardingLoaders.tsx:55]: Got it, thanks!

File: src/actions/onboardingActions.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (lalitkumarrock) [reply src/actions/onboardingActions.js:425]: Here calls multiple dispatch at a time. So difficult to manage the backoff retry process individually.

[the remark this answers is not in the document]
Comment (IAmKio) [reply src/actions/onboardingActions.js:425]: Just picking this up again @lalitkumarrock  - we're going to need to deal with this as we can't have users bring stuck here in any way as first impressions count. This breaking in any way due to network issues or server errors will need to either retry or restart the process. I'll leave it up to you to decide the best course of action but users can't get stuck here for sure.

[the remark this answers is not in the document]
Comment (lalitkumarrock) [reply src/actions/onboardingActions.js:425]: Okay, I'll find out the best solution

[the remark this answers is not in the document]
Comment (lalitkumarrock) [reply src/actions/onboardingActions.js:426]: @IAmKio Please review it. https://github.com/pillarwallet/pillarwallet/pull/3319/commits/e86a7f44377aae67d9216599de916c2b7f0a9387

https://github.com/pillarwallet/pillarwallet/assets/82652040/621f526c-b4cb-4b57-856c-9ed000fb35af

[the remark this answers is not in the document]
Comment (IAmKio) [reply src/actions/onboardingActions.js:426]: Great work @lalitkumarrock - as we just discussed, if you could change the "No internet" to "No internet, retrying..." to indicate to the user that we're handling it. Thank you

[the remark this answers is not in the document]
Comment (lalitkumarrock) [reply src/actions/onboardingActions.js:426]: Text updated.

**Pull request merged**